spi_nand_flash/src/nand_impl.c

```diff
@@ -232,16 +229,13 @@ esp_err_t nand_is_free(spi_nand_flash_device_t *handle, uint32_t page, bool *is_
     uint16_t used_marker;
 
     ESP_GOTO_ON_ERROR(read_page_and_wait(handle, page, NULL), fail, TAG, "");
-    ESP_GOTO_ON_ERROR(spi_nand_read(handle->config.device_handle, (uint8_t *)&used_marker,
+    ESP_GOTO_ON_ERROR(spi_nand_read(handle->config.device_handle, (uint8_t *)handle->read_buffer,
                                     handle->chip.page_size + 2, 2),
                       fail, TAG, "");
 
+    used_marker = *(uint16_t *)handle->read_buffer;
```

Comment:
```suggestion
    memcpy(&used_marker, handle->read_buffer, sizeof(used_marker));
```
read_buffer is probably going to be 4-byte aligned, but the compiler doesn't know this. So it's probably better to use an explicit memcpy.

(Same in the other case you have fixed.)

Reply:
Updated. Thank you!